cmake/Platform/Common/Install_common.cmake

```diff
@@ -430,6 +430,21 @@ function(ly_setup_cmake_install)
         DESTINATION .
         COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
     )
+    string(CONFIGURE [=[
+if("${CMAKE_INSTALL_CONFIG_NAME}" MATCHES "^([Rr][Ee][Ll][Ee][Aa][Ss][Ee])$")
+    set(install_output_folder "${CMAKE_INSTALL_PREFIX}/@runtime_output_directory@")
+    file(WRITE ${install_output_folder}/engine.json 
+"{
+    \"engine_name\": \"@LY_VERSION_ENGINE_NAME@\"
+}")
```

Comment:
Should we give a name other than `LY_VERSION_ENGINE_NAME`?  Could be anything, maybe there's benefit to giving a somewhat unique name based on the project's name so that there's less likelihood that it would be used by other O3DE things on a user's system.

Also, do we need to generate an `"engine"` field in the project.json that matches this?

Reply:
We don't need the "engien" field in the project.json file, because the [FindEngineRoot](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp#L272-L277) function wouldn't be able to locate a registered engine anyway on an end user machine, because their is no ~/.o3de/o3de_manifest.json in that case.

Furthermore because the engine.json file `exists` it will use that location as the engine root.

Ideally though we shouldn't actually need the Engine Root to be set and therefore have an engine.json in the GameLauncher, since the Game shouldn't be trying to access any files in the Engine Root itself

Reply:
@lumberyard-employee-dm ?
The engine name is IMO irrelevant, nothing should be reading that, I think we just need those files to be there. I put `LY_VERSION_ENGINE_NAME` because that is the name we use to install the engine

Reply:
One thing we could do as a fallback, is to have the `SettingRegistryMergeUtils::FindEngineRoot` function return the Project path if an engine path couldn't be reconcile, but TO NOT set the "/Amazon/AzCore/Bootstrap/engine_path" key.
For example we could update the end of `FindEngineRoot` to be
```
// Use the project.json and engine manifest to locate the engine root.
if (engineRoot = Internal::ReconcileEngineRootFromProjectPath(settingsRegistry, projectRoot); !engineRoot.empty())
{
    settingsRegistry.Set(engineRootKey, engineRoot.c_str());
    return engineRoot;
}

return projectRoot;
```

Reply:
can we do something similar for `project.json`? would be great to do it for both and not need those files in the output

Reply:
Well we need to locate the project root somehow.

Potential ways to do this is to scan upwards for a "Cache" directory use that as a fallback.

If we want to be less flaky, see if such a "Cache" directory contains a `<asset-type>/engine.pak` file within it and assume it is the project root(though at the point I think it should be named "project.pak").

Reply:
I am ok with the option of checking for a `engine.pak`, we can also do `project.pak` and just generate an empty one for the time being...
long term, will the asset bundler pak engine assets into the `engine.pak` and project assets into `project.pak`? if not, there is no real point of having those two

Reply:
I didn't mean to have to two pak files.
It was a minor observation that the engine.pak file could be renamed to project.pak.

It seems strange to me that we would detect the "project" root based on the location of a file called "engine.pak".
This is only a very minor nitpick from me.